### PR TITLE
Improvements from KvK and NYP

### PR DESCRIPTION
**Pull request by jvvbarbosaCND**
### Summary :memo:
A few improvement suggestions based on the work done at KVK and NYP in the PubSub trigger function.

### Details
1. Activate the correct APIs for the PubSub triggered function of gen2
2. Remove function zone distinction
3. Add secret manager access to the function's service account
4. Turn mandatory scheduler message into a JSON friendly string
5. Apply the `terraform fmt` command

### Follow-up
_Follows here instead of in an issue, since I can't seem to find a way to create an issue for a template repo._

Some of the modifications made here should actually be done in a central module that implements the basic necessities for deploying a gen2 cloud function (such as service activation and iam permissions). I believe that the secret manager permissions could be created also in such a module, under the conditional `enable_secret_access` or something. The reason why that is not done is because I'd like to do it properly and also give the module user a choice to create a new service account instead of using the default one. This should be easy to do on a conditional basis based on an input variables `service_account_name` and `create_service_account`. But i'd like to do that another day ;)

Also after this is merged I will post another PR (if necessary) for the improvements coming from the NYP project. I believe they are related and therefore there might be some overlap.

### Budget
Time spent on this PR review should be allocated in the following Harvest project: [CND42].

### CODE REVIEW

**Comment by jvvbarbosaCND** [comment]
Now also contains improvements from NYP  work

File: terraform/modules/gcf_gen2_pubsub_source_repo/variables.tf [hunk 1]
```diff
@@ -1,78 +1,80 @@
...
-  default = "europe-west1"
+  default     = "europe-west1"
 }
 
 variable "runtime" {
   description = "Runtime where function is operating."
-  type = string
-  default = "python310"
+  type        = string
+  default     = "python310"
```
Comment (dennisCND) [inline terraform/modules/gcf_gen2_pubsub_source_repo/variables.tf]: Do we want to change this default?

**Review by dennisCND — APPROVED** [review]
Looks good! we can discuss default python version more extensively at some point, but in general makes sense like this

File: terraform/modules/gcf_gen2_pubsub_source_repo/variables.tf (hunk 1 above)
Comment (jvvbarbosaCND) [reply terraform/modules/gcf_gen2_pubsub_source_repo/variables.tf]: I forgot about this point, so thanks for calling it out!

**Pull request merged**